Fix MinGW-w64/Clang `unknown-pragmas` warning.

### Recreation
`tree`
![Image](https://github.com/user-attachments/assets/c1ae32af-01f9-4f8a-8eea-ef4b9958b3dc)
`test.cpp`
```cpp
#include <vdf_parser.hpp>

int main(void) { return 0; }

```
`CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.30)
project(test CXX)

include(FetchContent)
FetchContent_Declare(ValveFileVDF GIT_REPOSITORY https://github.com/TinyTinni/ValveFileVDF.git GIT_TAG v1.1.0)
FetchContent_MakeAvailable(ValveFileVDF)

add_executable(test source/test.cpp)
target_link_libraries(test PRIVATE ValveFileVDF)
target_compile_options(test PRIVATE -Wall -Wextra)
```
### Result
`Configuration`
```
"C:\Program Files\JetBrains\CLion 2024.3.2\bin\cmake\win\x64\bin\cmake.exe" -DCMAKE_BUILD_TYPE=Debug "-DCMAKE_MAKE_PROGRAM=C:/Program Files/JetBrains/CLion 2024.3.2/bin/ninja/win/x64/ninja.exe" -G Ninja -S C:\Users\USERNAME\CLionProjects\test -B C:\Users\USERNAME\CLionProjects\test\cmake-build-debug
-- The CXX compiler identification is GNU 13.1.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/JetBrains/CLion 2024.3.2/bin/mingw/bin/g++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Cloning into 'valvefilevdf-src'...
HEAD is now at 63176a3 adds some wchar_t proptests (#29)
-- The C compiler identification is GNU 13.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/JetBrains/CLion 2024.3.2/bin/mingw/bin/gcc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done (2.3s)
-- Generating done (0.0s)
-- Build files have been written to: C:/Users/USERNAME/CLionProjects/test/cmake-build-debug

[Finished]
```
`Compilation`
```
====================[ Build | test | Debug ]====================================
"C:\Program Files\JetBrains\CLion 2024.3.2\bin\cmake\win\x64\bin\cmake.exe" --build C:\Users\USERNAME\CLionProjects\test\cmake-build-debug --target test -j 22
[1/2] Building CXX object CMakeFiles/test.dir/source/test.cpp.obj
In file included from C:/Users/USERNAME/CLionProjects/test/source/test.cpp:1:
C:/Users/USERNAME/CLionProjects/test/cmake-build-debug/_deps/valvefilevdf-src/include/vdf_parser.hpp:115: warning: ignoring '#pragma warning ' [-Wunknown-pragmas]
  115 | #pragma warning(push)
      | 
C:/Users/USERNAME/CLionProjects/test/cmake-build-debug/_deps/valvefilevdf-src/include/vdf_parser.hpp:116: warning: ignoring '#pragma warning ' [-Wunknown-pragmas]
  116 | #pragma warning(disable : 4996)
      | 
C:/Users/USERNAME/CLionProjects/test/cmake-build-debug/_deps/valvefilevdf-src/include/vdf_parser.hpp:122: warning: ignoring '#pragma warning ' [-Wunknown-pragmas]
  122 | #pragma warning(pop)
      | 
[2/2] Linking CXX executable test.exe

Build finished

```

I works 👍.

Although the following warnings are still present using `/Wall` on MSVC.
```
====================[ Build | test | Debug ]====================================
"C:\Program Files\JetBrains\CLion 2024.3.2\bin\cmake\win\x64\bin\cmake.exe" --build C:\Users\MichaelsoftBinbows\CLionProjects\test\cmake-build-debug --target test -j 22
[1/2] Building CXX object CMakeFiles\test.dir\source\test.cpp.obj
C:\Users\MichaelsoftBinbows\CLionProjects\test\cmake-build-debug\_deps\valvefilevdf-src\include\vdf_parser.hpp(93): warning C4514: 'tyti::vdf::detail::literal_macro_help<wchar_t>::result': unreferenced inline function has been removed
C:\Users\MichaelsoftBinbows\CLionProjects\test\cmake-build-debug\_deps\valvefilevdf-src\include\vdf_parser.hpp(98): warning C4514: 'tyti::vdf::detail::literal_macro_help<wchar_t>::result': unreferenced inline function has been removed
C:\Users\MichaelsoftBinbows\CLionProjects\test\cmake-build-debug\_deps\valvefilevdf-src\include\vdf_parser.hpp(106): warning C4514: 'tyti::vdf::detail::string_converter': unreferenced inline function has been removed
C:\Users\MichaelsoftBinbows\CLionProjects\test\cmake-build-debug\_deps\valvefilevdf-src\include\vdf_parser.hpp(108): warning C4514: 'tyti::vdf::detail::string_converter': unreferenced inline function has been removed
C:\Users\MichaelsoftBinbows\CLionProjects\test\cmake-build-debug\_deps\valvefilevdf-src\include\vdf_parser.hpp(238): warning C4514: 'tyti::vdf::Options::Options': unreferenced inline function has been removed
C:\Users\MichaelsoftBinbows\CLionProjects\test\cmake-build-debug\_deps\valvefilevdf-src\include\vdf_parser.hpp(248): warning C4514: 'tyti::vdf::WriteOptions::WriteOptions': unreferenced inline function has been removed
[2/2] Linking CXX executable test.exe

Build finished
```

--- a/include/vdf_parser.hpp
+++ b/include/vdf_parser.hpp
@@ -111,14 +111,14 @@ inline std::string string_converter(const std::wstring &w) NOEXCEPT
     auto wstr = w.data();
 // unsafe: ignores any error handling
 // and disables warning that wcsrtombs_s should be used
-#ifdef WIN32
+#ifdef _MSC_VER
 #pragma warning(push)
 #pragma warning(disable : 4996)
 #endif
     std::size_t len = 1 + std::wcsrtombs(nullptr, &wstr, 0, &state);
     std::string mbstr(len, '\0');
     std::wcsrtombs(&mbstr[0], &wstr, mbstr.size(), &state);
-#ifdef WIN32
+#ifdef _MSC_VER
 #pragma warning(pop)
 #endif
     return mbstr;

